src/lcm/LifeCycleActions.cc

```diff
@@ -1011,14 +1054,23 @@ void LifeCycleManager::clean_up_vm(VirtualMachine * vm, bool dispose,
 
     hpool->del_capacity(vm->get_hid(), sr);
 
-    const VectorAttribute * graphics = vm->get_template_attribute("GRAPHICS");
+    vm->release_vnc_port();
 
-    if ( graphics != 0 && (graphics->vector_value("PORT", port) == 0))
+    if (state == VirtualMachine::POWEROFF ||
+            state == VirtualMachine::SUSPENDED)
     {
-        clpool->release_vnc_port(vm->get_cid(), port);
-    }
+        float memory, cpu;
 
-    switch (state)
+        vm->get_template_attribute("MEMORY", memory);
+        vm->get_template_attribute("CPU", cpu);
+
+        quota_tmpl.add("RUNNING_MEMORY", -memory);
+        quota_tmpl.add("RUNNING_CPU", -cpu);
+        quota_tmpl.add("RUNNING_VMS", -1);
+
+        vmm->trigger_cleanup(vid, false);
+    }
+    else switch (lcmstate)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The switch statement here is missing a default statement. The default statement will catch any case values that are not explicitly handled by a case statement.  This prevents bugs if a case value is accidentally forgotten or new enum values are added later